public/app/features/annotations/executeAnnotationQuery.ts

```diff
@@ -23,7 +23,11 @@ export function executeAnnotationQuery(
     ...datasource.annotations,
   };
 
-  const annotation = processor.prepareAnnotation!(savedJsonAnno);
+  const annotationWithDefaults = {
+    ...processor.getDefaultQuery?.(),
```

Comment:
i'm not sure if running `getDefaultQuery` is necessary here. wouldn't the query already have the defaults applied when creating a new annotation query?

Reply:
I can see why it would be expected to have the defaults applied, yes. However, the defaults are applied only in the render() method in StandardAnnotationQueryEditor, after which the onChange isn't called. This is why it has to be applied here too, since if you run "test query" right after mount, it won't have the default values. I don't love the solution, especially if it's confusing to someone else seeing it. 
However, I haven't found a good place to do it: 
1. It can be left to the datasource devs to do it prepareAnnotation method in each datasource, but this seems a bit hidden, somehow?
2. Related to that, I could do it in verifyDatasource before prepareAnnotation is called, but neither runs when changing datasource (so it cant update with default values), only when changing the query. Not sure why, as I would expect it to, but it's due to how the code is structured and changing it isn't in the scope of this PR. But if you think it should be done before this is merged, let me know and we can check it out!

Reply:
i see. i think it would be worth looking into seeing if it could be run somehow during `verifyDatasource`. looks like it's called on mount and updates so should cover initial render and changing datasources.

Reply:
I agree, it's worth looking into. Will merge this now, but I will schedule a time to take a look at why componentDidUpdate doesn't trigger the verifyDatasource on datasource change